src/NzbDrone.Core.Test/Download/DownloadClientTests/PorlaTests/PorlaFixture.cs

```diff
@@ -0,0 +1,595 @@
+            //     SavePath = "/tmp",
+            //     SeedingDuration = 0L,
+            //     Session = "default",
+            //     Size = 100000000L,
+            //     State = LibTorrentStatus.downloading,  // ?
+            //     Tags = new (SomeTags),
+            //     Total = 100000000L,
+            //     TotalDone = 150000000L,
+            //     UploadRate = 100000
+            // };
+
+            _seeding = new PorlaTorrentDetail
+            {
+                ActiveDuration = 120L,
+                AllTimeDownload = 200000000L,
+                AllTimeUpload = 100000000L,
+                Category = "sonarr-tv",
+                DownloadRate = 100,  // this seems to always be doing something / might be frozen at last value
+                Error = null,
+                ETA = -1L,  // we are done so eta is infinate
+                FinishedDuration = 100L,
+                Flags = new (DefaultTags),
+                InfoHash = new (_somehash, null),
+                LastDownload = 100L,
+                LastUpload = 10L,
+                ListPeers = 128,
+                ListSeeds = 16,
+                Metadata = null,
+                MovingStorage = false,
+                Name = _title,
+                NumPeers = 1,
+                NumSeeds = 2,
+                Progress = 1.0f,
+                QueuePosition = -1,
+                Ratio = 1.0d,
+                SavePath = "/tmp",
+                SeedingDuration = 666L,
+                Session = "default",
+                Size = 190000000L,  // usually a little smaller than downloaded total
+                State = LibTorrentStatus.seeding,  // double check this one
+                Tags = new (SomeTags),  // do we have a remote episode aviable? can I use CreateRemoteEpisode
+                Total = 0L,
+                TotalDone = 190000000L, // after we are finished this should be the same as `size`
+                UploadRate = 100
+            };
+
+            Mocker.GetMock<ITorrentFileInfoReader>()
+                  .Setup(s => s.GetHashFromTorrentFile(It.IsAny<byte[]>()))
+                  .Returns(_somehash);
+
+            Mocker.GetMock<IHttpClient>()
+                  .Setup(s => s.Get(It.IsAny<HttpRequest>()))
+                  .Returns<HttpRequest>(r => new HttpResponse(r, new HttpHeader(), Array.Empty<byte>()));
+        }
+
+        /// <summary> Setups the `Add*` to fail, by throwing Exceptions </summary>
+        protected void SetupFailedDownload()
+        {
+            Mocker.GetMock<IPorlaProxy>()
+                .Setup(s => s.AddMagnetTorrent(It.IsAny<PorlaSettings>(), It.IsAny<string>(), It.IsAny<IList<string>>()))
+                .Throws<InvalidOperationException>();
+
+            Mocker.GetMock<IPorlaProxy>()
+                .Setup(s => s.AddTorrentFile(It.IsAny<PorlaSettings>(), It.IsAny<byte[]>(), It.IsAny<IList<string>>()))
+                .Throws<InvalidOperationException>();
+
+            #nullable enable
```

Comment:
These should not be necessary.

Reply:
You would think so!

I think there there is something wrong between the JSONRPC decoding json nulls to C# objects that I should check somewhere. 

I get extremely confusing errors if I don't include the nullable flag